libs/ml/src/ops/matrix_multiply.cpp

```diff
@@ -285,6 +285,25 @@ std::vector<typename fetch::math::SizeType> MatrixMultiply<T>::ComputeOutputShap
   return output_shape;
 }
 
+template <typename T>
+MLChargeAmount MatrixMultiply<T>::ChargeForward()
+{
+  assert(!this->batch_input_shapes_.empty());
+
+  // TODO(VH): impl. for n-dimensional case, not only for 2D.
+  assert(this->batch_input_shapes_.size() == 2);
+
+  // Assuming this is a matrix multiplication of weights * input_vector
```

Comment:
if you take a look at math/matrix_operations.hpp you can see that the Dot() method (which is what matmul operation invokes) uses blas/gemm_nn; so if we look at gemm_nn_novector.cpp we can count the number of operations in the non vectorised implementation. It looks to me like this should be something like:

resut.size() + n.width() * (1 + (result.height() * 2))
I haven't factored in assignment operations, but the compiler would probably remove some of these, anyway it's probably slightly off.

Reply:
Here I use a worst-case estimation which is for A[n;m] dot B[m;p] == O(n * m * p), or O(n^3) if the matrices are square and of size n. I know there are more effective algorithms, but at this stage I wouldn't dive into particular implementation, we will need to benchmark/measure it anyway.

Reply:
okay we can leave it for next iteration then